Build fails looking for ares.h on luvit 0.7.0

git submodule update --init deps/hiredis
Submodule 'deps/hiredis' (git://github.com/redis/hiredis.git) registered for path 'deps/hiredis'
Cloning into 'deps/hiredis'...
remote: Counting objects: 2165, done.
remote: Compressing objects: 100% (958/958), done.
remote: Total 2165 (delta 1288), reused 2012 (delta 1158)
Receiving objects: 100% (2165/2165), 653.51 KiB | 0 bytes/s, done.
Resolving deltas: 100% (1288/1288), done.
Checking connectivity... done.
Submodule path 'deps/hiredis': checked out 'bcf83543762577a9b0f5625c848b7061cc980f30'
make -C deps/hiredis libhiredis.a
make[1]: Entering directory `/root/luvit-redis/deps/hiredis'
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  net.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  hiredis.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  sds.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  async.c
ar rcs libhiredis.a net.o hiredis.o sds.o async.o
make[1]: Leaving directory`/root/luvit-redis/deps/hiredis'
mkdir -p build
cc -I/usr/local/include/luvit -I/usr/local/include/luvit/uv -I/usr/local/include/luvit/luajit -I/usr/local/include/luvit/http_parser -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -Wall  -fPIC -Ideps/hiredis -Isrc -o build/redis.luvit src/redis.c deps/hiredis/libhiredis.a -shared -lm deps/hiredis/libhiredis.a
In file included from src/redis.c:26:0:
src/luvit-hiredis-adapter.h:15:5: error: unknown type name 'ev_io'
     ev_io rev, wev;
     ^
src/luvit-hiredis-adapter.h:18:33: error: unknown type name 'EV_P_'
 static void redisLibevReadEvent(EV_P_ ev_io _watcher, int revents) {
                                 ^
src/luvit-hiredis-adapter.h:28:34: error: unknown type name 'EV_P_'
 static void redisLibevWriteEvent(EV_P_ ev_io *watcher, int revents) {
                                  ^
src/luvit-hiredis-adapter.h: In function 'redisLibevAddRead':
src/luvit-hiredis-adapter.h:44:9: warning: implicit declaration of function 'ev_io_start' [-Wimplicit-function-declaration]
         ev_io_start(EV_A_ &e->rev);
         ^
src/luvit-hiredis-adapter.h:44:21: error: 'EV_A_' undeclared (first use in this function)
         ev_io_start(EV_A_ &e->rev);
                     ^
src/luvit-hiredis-adapter.h:44:21: note: each undeclared identifier is reported only once for each function it appears in
src/luvit-hiredis-adapter.h: In function 'redisLibevDelRead':
src/luvit-hiredis-adapter.h:54:9: warning: implicit declaration of function 'ev_io_stop' [-Wimplicit-function-declaration]
         ev_io_stop(EV_A_ &e->rev);
         ^
src/luvit-hiredis-adapter.h:54:20: error: 'EV_A_' undeclared (first use in this function)
         ev_io_stop(EV_A_ &e->rev);
                    ^
src/luvit-hiredis-adapter.h: In function 'redisLibevAddWrite':
src/luvit-hiredis-adapter.h:64:21: error: 'EV_A_' undeclared (first use in this function)
         ev_io_start(EV_A_ &e->wev);
                     ^
src/luvit-hiredis-adapter.h: In function 'redisLibevDelWrite':
src/luvit-hiredis-adapter.h:74:20: error: 'EV_A_' undeclared (first use in this function)
         ev_io_stop(EV_A_ &e->wev);
                    ^
src/luvit-hiredis-adapter.h: At top level:
src/luvit-hiredis-adapter.h:85:29: error: unknown type name 'EV_P_'
 static int redisLibevAttach(EV_P_ redisAsyncContext *ac) {
                             ^
src/redis.c: In function 'lua_create_client':
src/redis.c:359:3: warning: implicit declaration of function 'redisLibevAttach' [-Wimplicit-function-declaration]
   redisLibevAttach(EV_DEFAULT_ redis_async_context);
   ^
src/redis.c:359:20: error: 'EV_DEFAULT_' undeclared (first use in this function)
   redisLibevAttach(EV_DEFAULT_ redis_async_context);
                    ^
src/redis.c:359:32: error: expected ')' before 'redis_async_context'
   redisLibevAttach(EV_DEFAULT_ redis_async_context);
                                ^
In file included from src/redis.c:26:0:
src/redis.c: At top level:
src/luvit-hiredis-adapter.h:38:13: warning: 'redisLibevAddRead' defined but not used [-Wunused-function]
 static void redisLibevAddRead(void *privdata) {
             ^
src/luvit-hiredis-adapter.h:58:13: warning: 'redisLibevAddWrite' defined but not used [-Wunused-function]
 static void redisLibevAddWrite(void *privdata) {
             ^
src/luvit-hiredis-adapter.h:78:13: warning: 'redisLibevCleanup' defined but not used [-Wunused-function]
 static void redisLibevCleanup(void *privdata) {
             ^
make: *_\* [build/redis.luvit] Error 1

--- a/src/luvit-hiredis-adapter.h
+++ b/src/luvit-hiredis-adapter.h
@@ -1,119 +1,124 @@
-#ifndef __HIREDIS_LIBEV_H__
-#define __HIREDIS_LIBEV_H__
+#ifndef __HIREDIS_LIBUV_H__
+#define __HIREDIS_LIBUV_H__
 
 #include <stdlib.h>
-#include <sys/types.h>
+#include <string.h>
 
 #include "uv.h"
 #include "hiredis.h"
 #include "async.h"
 
-typedef struct redisLibevEvents {
-    redisAsyncContext *context;
-    struct ev_loop *loop;
-    int reading, writing;
-    ev_io rev, wev;
-} redisLibevEvents;
 
-static void redisLibevReadEvent(EV_P_ ev_io *watcher, int revents) {
-#if EV_MULTIPLICITY
-    ((void)loop);
-#endif
-    ((void)revents);
+typedef struct redisLibuvEvents {
+  redisAsyncContext* context;
+  uv_poll_t          handle;
+  int                events;
+} redisLibuvEvents;
 
-    redisLibevEvents *e = (redisLibevEvents*)watcher->data;
-    redisAsyncHandleRead(e->context);
+
+static void redisLibuvPoll(uv_poll_t* handle, int status, int events) {
+  redisLibuvEvents* p = (redisLibuvEvents*)handle->data;
+
+  if (status != 0) {
+    return;
+  }
+
+  if (events & UV_READABLE) {
+    redisAsyncHandleRead(p->context);
+  }
+  if (events & UV_WRITABLE) {
+    redisAsyncHandleWrite(p->context);
+  }
 }
 
-static void redisLibevWriteEvent(EV_P_ ev_io *watcher, int revents) {
-#if EV_MULTIPLICITY
-    ((void)loop);
-#endif
-    ((void)revents);
 
-    redisLibevEvents *e = (redisLibevEvents*)watcher->data;
-    redisAsyncHandleWrite(e->context);
+static void redisLibuvAddRead(void *privdata) {
+  redisLibuvEvents* p = (redisLibuvEvents*)privdata;
+
+  p->events |= UV_READABLE;
+
+  uv_poll_start(&p->handle, p->events, redisLibuvPoll);
 }
 
-static void redisLibevAddRead(void *privdata) {
-    redisLibevEvents *e = (redisLibevEvents*)privdata;
-    struct ev_loop *loop = e->loop;
-    ((void)loop);
-    if (!e->reading) {
-        e->reading = 1;
-        ev_io_start(EV_A_ &e->rev);
-    }
+
+static void redisLibuvDelRead(void *privdata) {
+  redisLibuvEvents* p = (redisLibuvEvents*)privdata;
+
+  p->events &= ~UV_READABLE;
+
+  if (p->events) {
+    uv_poll_start(&p->handle, p->events, redisLibuvPoll);
+  } else {
+    uv_poll_stop(&p->handle);
+  }
 }
 
-static void redisLibevDelRead(void *privdata) {
-    redisLibevEvents *e = (redisLibevEvents*)privdata;
-    struct ev_loop *loop = e->loop;
-    ((void)loop);
-    if (e->reading) {
-        e->reading = 0;
-        ev_io_stop(EV_A_ &e->rev);
-    }
+
+static void redisLibuvAddWrite(void *privdata) {
+  redisLibuvEvents* p = (redisLibuvEvents*)privdata;
+
+  p->events |= UV_WRITABLE;
+
+  uv_poll_start(&p->handle, p->events, redisLibuvPoll);
 }
 
-static void redisLibevAddWrite(void *privdata) {
-    redisLibevEvents *e = (redisLibevEvents*)privdata;
-    struct ev_loop *loop = e->loop;
-    ((void)loop);
-    if (!e->writing) {
-        e->writing = 1;
-        ev_io_start(EV_A_ &e->wev);
-    }
+
+static void redisLibuvDelWrite(void *privdata) {
+  redisLibuvEvents* p = (redisLibuvEvents*)privdata;
+
+  p->events &= ~UV_WRITABLE;
+
+  if (p->events) {
+    uv_poll_start(&p->handle, p->events, redisLibuvPoll);
+  } else {
+    uv_poll_stop(&p->handle);
+  }
 }
 
-static void redisLibevDelWrite(void *privdata) {
-    redisLibevEvents *e = (redisLibevEvents*)privdata;
-    struct ev_loop *loop = e->loop;
-    ((void)loop);
-    if (e->writing) {
-        e->writing = 0;
-        ev_io_stop(EV_A_ &e->wev);
-    }
+
+static void on_close(uv_handle_t* handle) {
+  redisLibuvEvents* p = (redisLibuvEvents*)handle->data;
+
+  free(p);
 }
 
-static void redisLibevCleanup(void *privdata) {
-    redisLibevEvents *e = (redisLibevEvents*)privdata;
-    redisLibevDelRead(privdata);
-    redisLibevDelWrite(privdata);
-    free(e);
+
+static void redisLibuvCleanup(void *privdata) {
+  redisLibuvEvents* p = (redisLibuvEvents*)privdata;
+
+  uv_close((uv_handle_t*)&p->handle, on_close);
 }
 
-static int redisLibevAttach(EV_P_ redisAsyncContext *ac) {
-    redisContext *c = &(ac->c);
-    redisLibevEvents *e;
 
-    /* Nothing should be attached when something is already attached */
-    if (ac->ev.data != NULL)
-        return REDIS_ERR;
+int redisLibuvAttach(redisAsyncContext* ac, uv_loop_t* loop) {
+  redisContext *c = &(ac->c);
 
-    /* Create container for context and r/w events */
-    e = (redisLibevEvents*)malloc(sizeof(*e));
-    e->context = ac;
-#if EV_MULTIPLICITY
-    e->loop = loop;
-#else
-    e->loop = NULL;
-#endif
-    e->reading = e->writing = 0;
-    e->rev.data = e;
-    e->wev.data = e;
+  if (ac->ev.data != NULL) {
+    return REDIS_ERR;
+  }
 
-    /* Register functions to start/stop listening for events */
-    ac->ev.addRead = redisLibevAddRead;
-    ac->ev.delRead = redisLibevDelRead;
-    ac->ev.addWrite = redisLibevAddWrite;
-    ac->ev.delWrite = redisLibevDelWrite;
-    ac->ev.cleanup = redisLibevCleanup;
-    ac->ev.data = e;
+  ac->ev.addRead  = redisLibuvAddRead;
+  ac->ev.delRead  = redisLibuvDelRead;
+  ac->ev.addWrite = redisLibuvAddWrite;
+  ac->ev.delWrite = redisLibuvDelWrite;
+  ac->ev.cleanup  = redisLibuvCleanup;
 
-    /* Initialize read/write events */
-    ev_io_init(&e->rev,redisLibevReadEvent,c->fd,EV_READ);
-    ev_io_init(&e->wev,redisLibevWriteEvent,c->fd,EV_WRITE);
-    return REDIS_OK;
+  redisLibuvEvents* p = malloc(sizeof(*p));
+
+  if (!p) {
+    return REDIS_ERR;
+  }
+
+  memset(p, 0, sizeof(*p));
+
+  if (uv_poll_init(loop, &p->handle, c->fd) != 0) {
+    return REDIS_ERR;
+  }
+
+  ac->ev.data    = p;
+  p->handle.data = p;
+  p->context     = ac;
+
+  return REDIS_OK;
 }
-
 #endif

--- a/src/redis.c
+++ b/src/redis.c
@@ -356,7 +356,7 @@ static int lua_create_client(lua_State *L)
     return luaL_error(L, redis_async_context->errstr);
   }
 
-  redisLibevAttach(EV_DEFAULT_ redis_async_context);
+  redisLibuvAttach(redis_async_context, uv_default_loop());
 
   lua_redis_client = (lua_redis_client_t *)lua_newuserdata(
                                           L, sizeof(lua_redis_client_t));

